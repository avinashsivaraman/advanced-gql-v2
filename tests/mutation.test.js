const gql = require('graphql-tag')
const createTestServer = require('./helper')
const CREATE_POST = gql`
  mutation {
    createPost(input: { message: "hello" }) {
      message
    }
  }
`

const UPDATE_SETTING = gql`
  mutation {
    updateSettings(input: { theme: DARK }) {
      id
      theme
    }
  }
`

describe('Mutation', () => {
  test('CreatePost', async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          createOne: jest.fn(() => [
            {
              id: 1,
              message: 'hello',
              createdAt: 12345839,
              likes: 20,
              views: 300,
            },
          ]),
        },
      },
    })

    const res = await mutate({ query: CREATE_POST })
    expect(res).toMatchSnapshot()
  })
  test('UpdateSettings', async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Settings: {
          updateOne(_, input) {
            return {
              id: 1,
              ...input,
            }
          },
        },
      },
    })
    const res = await mutate({ query: UPDATE_SETTING })
    expect(res.data.updateSettings).toEqual({ id: '1', theme: 'DARK' })
  })
})
