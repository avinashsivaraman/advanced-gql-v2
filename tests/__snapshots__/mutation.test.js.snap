// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation CreatePost 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "message": "Hello World",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
